addons/spreadsheet/static/src/pivot/pivot_model.js

```diff
@@ -514,6 +514,21 @@ export class OdooPivotModel extends PivotModel {
         });
     }
 
+    /**
+     * @override
+     */
+    _sanitizeLabel(value, groupBy, config) {
+        const { metaData } = config;
+        const fieldName = groupBy.split(":")[0];
+        if (fieldName && metaData.fields[fieldName]) {
+            const fields = fieldName.split(".");
+            if (fields.length > 1 && fields.at(-1) === "id" && Array.isArray(value)) {
```

Comment:
even more straight forward ?
but maybe we don't want to wait for another full runbot cycle
```suggestion
            if (fieldName.endsWith(".id") && Array.isArray(value)) {
```